Add target support for MSP432E4 series

9names opened the pull request:
Adding support for MSP432E4 targets (MSP432E401Y and MSP432E411Y) .
Generated via target-gen from https://software-dl.ti.com/msp430/msp430_public_sw/mcu/msp430/msp432cmsis/latest/exports/TexasInstruments.MSP432E4_DFP.3.2.6.pack

Tested with probe-rs-cli download and hs-probe - working!

~~Tested via the onboard XDS110 CMSIS-DAP compatible debugger on MSP-EXP432E401Y - not functional.
XDS110 also fails with known good targets (STM32H7, RP2040) so I expect the issue lies with the debugger, not the algo.~~

XDS110 actually works fine, but the probe defaults to JTAG which probe-rs doesn't support.
Works fine when specifying protocol as SWD

prabhpreet commented:
@9names Thanks for your help over the Rust-Embedded matrix channel. I found this PR and thought I'd comment here for future reference regarding XDS110 support.

When using `cargo-flash` I saw in the logs that `probe-rs` was using the JTAG over CMSIS-DAP protocol. I got confirmation from @Yatekii over the probe-rs matrix channel that JTAG over CMSIS-DAP is not supported yet. `cargo-embed` on the other hand uses SWD by default majority of the times, and works well with XDS110.

For `cargo-flash`, the argument I had to pass to force the protocol was `--protocol=swd`.

For `cargo-embed`, I had to append the following to `Embed.toml`:
```
[default.probe]
protocol="Swd"
```

9names commented:
You are correct, it does work when swd is specified on the command-line. fascinating.

```system
$ ./release/probe-rs-cli dump 0x0 4 --chip MSP432E411Y --protocol swd
Addr 0x00000000: 0x20040000
Addr 0x00000004: 0x000001fd
Addr 0x00000008: 0x000003d7
Addr 0x0000000c: 0x0000070b
Read 4 words in 3.998451ms
```

probe-rs tooling used to always default to SWD, which caught me out at first when targeting RISCV targets.
but now... it seems to depend on the probe which one is used? makes sense for my probes that *only* support SWD, but hsprobe supports both. I don't get it.

prabhpreet commented:
I haven't completely hammered down the specifics in the `probe-rs` source code but I think I have a clue. 
Through tracing some of the commands exchanged using the CMSIS-DAP protocol is `DAP_Connect` (https://arm-software.github.io/CMSIS_5/DAP/html/group__DAP__Connect.html)

When a protocol is not specified, the logs show sending a request to connect with the Default Port of the device. The response from XDS110 shows it selects the JTAG mode as default. But `probe-rs` doesn't have support for JTAG specific commands further down the road for JTAG mode. That's probably why it fails.

9names commented:
I've rebased this and updated this using the latest git target-gen since the target format has changed while this PR has been waiting for review.
Tested it again with the MSP432E411Y launchpad, still works fine.

Yatekii reviewed (approved):
Thanks a lot!
